### PR TITLE
Create new section "Additional Mechanisms for Storage Protection"

### PR DESCRIPTION
**Pull request by tomschr**
Title: Additional Mechanisms for Storage Protection

* Change content of sg_persist
  * Add remarks for developers
  * Add introducery paragraph
  * Add additional check if the disk supports persistent
    reservations
  * Combine the two procedure into one; move para inbetween
    to the end

* Correct sect1 -> sect2

---

Remaining ToDos:
* [x] reorganize sg_persist and sfex information in new section 'Additional Mechanisms...'
* [x] fix bsc#1090169
* [x] check and adjust all IDs
* [ ]  add section(s) for new feature (diskless SBD) 
* [ ] re-add step overview to 'Steps to Set Up Storage-based Protection' (after diskless SBD is in, too)

### CODE REVIEW

File: xml/ha_storage_protection.xml [hunk 1]
```diff
@@ -779,27 +782,67 @@ Timeout (msgwait)  : 10
...
+   &stonith; Block Devices (SBD), <literal>watchdog</literal> support and
+   the <literal>external/sbd</literal> &stonith; agent.
+  </para>
+ </sect1>-->
+
+ <sect1 xml:id="sec.ha.storage.protect.morestuff">
+  <title>Additional Mechanisms for Storage Protection</title>
+  <remark>toms 2018-04-20: this can be improved...</remark>
+  <para>Apart from &stonith;, there are other methods to reach storage
+  protection. SCSI-3 and SCSI-4 uses persistent reservations whereas
```
Comment (taroth21) [inline xml/ha_storage_protection.xml]: maybe switch to plural here: SCSI-3 and SCSI-4 use

File: xml/ha_storage_protection.xml [hunk 2]
```diff
@@ -779,27 +782,67 @@ Timeout (msgwait)  : 10
...
+  <sect2 xml:id="sec.ha.storage.protect.sgpersist">
    <title>Configuring an sg_persist Resource</title>
-   <remark>toms 2018-04-18: Not sure why this section is here. It does not
-    belong to the described procedure introduced in
-    <xref linkend="sec.ha.storageprotection.fencing.setup"/>.
-    I'm tempted to move it...</remark>
+   <remark>toms 2018-04-20: I would like to see a little bit more background
+   information. </remark>
+   <para>
+    The SCSI specification 3 and 4 defines <emphasis>persistent reservations</emphasis>.
```
Comment (taroth21) [inline xml/ha_storage_protection.xml]: same here: plural? The SCSI specification 3 and 4 define...

File: xml/ha_storage_protection.xml [hunk 3]
```diff
@@ -779,27 +782,67 @@ Timeout (msgwait)  : 10
...
      disk compatible. <literal>sg_persist</literal> only works for devices like
-     SCSI disks or iSCSI LUNs. Do <emphasis>not</emphasis> use it for IDE,
-     SATA, or any block devices which do not support the SCSI protocol. </para>
+     SCSI disks or iSCSI LUNs.
+     <remark>toms 2018-04-20: What about FCoE, FC, iSER, SRP, Serial Attached SCSI (SAR)?</remark>
+     Do <emphasis>not</emphasis> use it for IDE, SATA, or any block devices
+     which do not support the SCSI protocol. </para>
    </note>
-   <remark>toms 2014-09-10: FATE#312345</remark>
+   <para>Before you proceed, first check if your disk supports
```
Comment (taroth21) [inline xml/ha_storage_protection.xml]: remove 'first'?

File: xml/ha_storage_protection.xml [hunk 4]
```diff
@@ -779,27 +782,67 @@ Timeout (msgwait)  : 10
...
+    persistent reservations. Use the following command (replace
+     <replaceable>DISK</replaceable> with your device name):</para>
+    <screen>&prompt.root;<command>sg_persist</command> -n --in --read-reservation -d /dev/<replaceable>DISK</replaceable></screen>
+   <para>The result will show you if your disk is supports persistent
```
Comment (taroth21) [inline xml/ha_storage_protection.xml]: typo (and you can use present tense for results, to keep it simple): The result shows if your disk supports persistent reservations

File: xml/ha_storage_protection.xml [hunk 5]
```diff
@@ -779,27 +782,67 @@ Timeout (msgwait)  : 10
...
+   <para>If you get an error message (like the one above), replace the old
+    disk with a SCSI compatible disk. Otherwise proceed as follows:</para>
    <procedure>
     <step>
-     <para> Log in as &rootuser; and start a shell. </para>
-    </step>
-    <step>
-     <para> Run the following commands to create the primitive resources
-       <literal>sg_persist</literal>: </para>
+     <para> Run the following commands as &rootuser; to create the primitive
```
Comment (taroth21) [inline xml/ha_storage_protection.xml]: Name target first: To create the primitive resource, run...

File: xml/ha_storage_protection.xml [hunk 6]
```diff
@@ -844,146 +882,140 @@ Timeout (msgwait)  : 10
...
-  <sect2 xml:id="sec.ha.storageprotection.exstoract.requirements">
-   <title>Setup</title>
    <para>
-    In the following, learn how to create a shared partition for use with
-    sfex and how to configure a resource for the sfex lock in the CIB. A
-    single sfex partition can hold any number of locks, and needs 1 KB 
-    of storage space allocated per lock.
-    By default, sfex_init creates one lock on the partition.
+    By design, sfex cannot be used with workloads that require concurrency
+    (such as OCFS2), but serves as a layer of protection for classic failover
```
Comment (taroth21) [inline xml/ha_storage_protection.xml]: use entity: &ofcs;

File: xml/ha_storage_protection.xml [hunk 7]
```diff
@@ -844,146 +882,140 @@ Timeout (msgwait)  : 10
...
+   </sect3>
+
+   <sect3 xml:id="sec.ha.storage.protect.exstoract.requirements">
+    <title>Setup</title>
+    <para>
+     In the following, learn how to create a shared partition for use with
+     sfex and how to configure a resource for the sfex lock in the CIB. A
+     single sfex partition can hold any number of locks, and needs 1 KB
+     of storage space allocated per lock.
+     By default, sfex_init creates one lock on the partition.
```
Comment (taroth21) [inline xml/ha_storage_protection.xml]: ```<command>sfex_init</command>```

File: xml/ha_storage_protection.xml [hunk 8]
```diff
@@ -844,146 +882,140 @@ Timeout (msgwait)  : 10
...
-      the resource to be protected has the id
-      <literal>filesystem1</literal>:
-     </para>
-<screen>&prompt.crm.conf;<command>order</command> order-sfex-1 inf: sfex_1 filesystem1
+     </step>
+     <step>
+      <para>
+       To protect resources via an sfex lock, create mandatory ordering and
+       placement constraints between the protectees and the sfex resource. If
+       the resource to be protected has the id
```
Comment (taroth21) [inline xml/ha_storage_protection.xml]: id -> ID

**Comment by tomschr** [comment]
@taroth21 Thanks for the corrections! Committed to 8ed70d3f584.

**Pull request merged**